rskj-core/src/main/java/co/rsk/net/AsyncNodeBlockProcessor.java

```diff
@@ -97,18 +99,25 @@ public BlockProcessResult processBlock(@Nullable Peer sender, @Nonnull Block blo
             return ignoreBlockResult(block, start);
         }
 
-        // Check if block is ready for processing - if the block is not too advanced, its parent block is in place etc.
-        boolean readyForProcessing = blockSyncService.preprocessBlock(block, sender, false);
-        if (readyForProcessing) {
-            // Validate block if it can be added to the queue for processing
-            if (isBlockValid(block)) {
-                scheduleForProcessing(new BlockInfo(sender, block), blockNumber, blockHash, peer);
+        // Check if block is ready for processing - if the block is not too advanced, its ancestor blocks are in place etc.
+        List<Block> blocksToConnect = blockSyncService.preprocessBlock(block, sender, false);
+        if (!blocksToConnect.isEmpty()) {
+            // if there's only one block to connect without any ancestors, then schedule its processing ( if the block is valid ofc )
+            if (blocksToConnect.size() == 1 && blocksToConnect.get(0).getHash().equals(block.getHash())) {
```

Comment:
I'd extract this to a variable or method with a meaningful name, i.e: `readyToProcess`